themes/root/templates/RecordDriver/AbstractBase/export-marc.phtml

```diff
@@ -1 +1 @@
-<?=$this->driver->getMarcRecord()->toRaw()?>
+<?=$this->driver->getMarcReader()->toISO2709() ?: $this->driver->getMarcReader()->toXML() ?>
```

Comment:
What is the purpose of the fallback here? I don't think outputting XML in this context will work right because of the associated HTTP headers.

Reply:
There's a chance that the record can't be serialized in ISO2709. If a record has a field longer than 9999 characters or the record itself is more than 99999 characters, it doesn't fit in the constraints of ISO2709. We have some records like that. During testing I couldn't see this to cause other problems than the result being in unexpected format when exporting single records, but bulk export would indeed be problematic. We don't currently have a mechanism for handling the situation where the requested format cannot be provided after all. Throwing an exception or adding error handling that would check for an empty result could help. What do you think?

Reply:
Yes, I think throwing an exception could make sense here. Another option would be making the logic to detect which export methods are supported a bit more robust (though it may be too expensive to do a lot of analysis there).

Reply:
I added an exception and basic handling for it.